sycl/include/sycl/properties/image_properties.hpp

```diff
@@ -50,41 +54,30 @@ template <int Dimensions, typename AllocatorT> class sampled_image;
 template <int Dimensions, typename AllocatorT> class unsampled_image;
 
 // SYCL 1.2.1 image property trait specializations
-template <int Dimensions, typename AllocatorT>
-struct is_property_of<property::image::use_host_ptr,
-                      image<Dimensions, AllocatorT>> : std::true_type {};
-template <int Dimensions, typename AllocatorT>
-struct is_property_of<property::image::use_mutex, image<Dimensions, AllocatorT>>
-    : std::true_type {};
-template <int Dimensions, typename AllocatorT>
-struct is_property_of<property::image::context_bound,
-                      image<Dimensions, AllocatorT>> : std::true_type {};
+#define __SYCL_MANUALLY_DEFINED_PROP(NS_QUALIFIER, PROP_NAME)                  \
+  template <int Dimensions, typename AllocatorT>                               \
+      struct is_property_of < NS_QUALIFIER::PROP_NAME,                         \
+      image<Dimensions, AllocatorT> : std::true_type {};
+#define __SYCL_DATA_LESS_PROP(NS_QUALIFIER, PROP_NAME, ENUM_VAL)               \
+  __SYCL_MANUALLY_DEFINED_PROP(NS_QUALIFIER, PROP_NAME)
+#include <sycl/properties/image_properties.def>
 
 // SYCL 2020 image property trait specializations
-template <int Dimensions, typename AllocatorT>
-struct is_property_of<property::image::use_host_ptr,
-                      sampled_image<Dimensions, AllocatorT>> : std::true_type {
-};
-template <int Dimensions, typename AllocatorT>
-struct is_property_of<property::image::use_mutex,
-                      sampled_image<Dimensions, AllocatorT>> : std::true_type {
-};
-template <int Dimensions, typename AllocatorT>
-struct is_property_of<property::image::context_bound,
-                      sampled_image<Dimensions, AllocatorT>> : std::true_type {
-};
-template <int Dimensions, typename AllocatorT>
-struct is_property_of<property::image::use_host_ptr,
-                      unsampled_image<Dimensions, AllocatorT>>
-    : std::true_type {};
-template <int Dimensions, typename AllocatorT>
-struct is_property_of<property::image::use_mutex,
-                      unsampled_image<Dimensions, AllocatorT>>
-    : std::true_type {};
-template <int Dimensions, typename AllocatorT>
-struct is_property_of<property::image::context_bound,
-                      unsampled_image<Dimensions, AllocatorT>>
-    : std::true_type {};
+#define __SYCL_MANUALLY_DEFINED_PROP(NS_QUALIFIER, PROP_NAME)                  \
+  template <int Dimensions, typename AllocatorT>                               \
+      struct is_property_of < NS_QUALIFIER::PROP_NAME,                         \
+      sampled_image<Dimensions, AllocatorT> : std::true_type {};
```

Comment:
```suggestion
      sampled_image<Dimensions, AllocatorT>> : std::true_type {};
```

Reply:
yeah, thanks. I genially fixed that in build/include :-) 
thanks for confirmation that it is what you were suggested, PR is not fully ready, I need to implement the same for the other SYCL objects